src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,10 +27,23 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
+        String[] splitAddress = trimmedAddress.split(",");
+        int numParts = splitAddress.length;
+
         this.isPrivate = isPrivate;
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
+        this.block = new Block(splitAddress[0]);
```

Comment:
Better to define what "splitAddress[0]" etc, are by assigning them good variable names.